### ISSUE TITLE
New API - websocket https scheme

### ISSUE THREAD

**Issue by krychla1** [etherdelta/etherdelta.github.io#205]
Hi Zack,
firstly thank you for your efforts and hard work on this project, you do a great job.

I would like to move my C# code from old API to the new one. Is it possible to use use the websocket protocol? I am stuck on your ED host, it uses https scheme (instead of wss or ws).

Thank you.

BR,
Ivan

**Comment by zackcoburn** [comment]
It should be doable. Did you see the example bots? They may help you
translate into C#.

https://github.com/etherdelta/etherdelta.github.io/tree/master/bots

Zack Coburn

NOTICE: This e-mail transmission (and/or the attachments accompanying it)
may contain confidential information belonging to the sender which is
protected by law. The information is intended only for the use of the
intended recipient. If you are not the intended recipient, you are hereby
notified that any disclosure, copying, distribution or the taking of any
action in reliance on the contents of this information is strictly
prohibited. Any unauthorized interception of this transmission is illegal.
If you have received this transmission in error, please promptly notify the
sender by reply e-mail, and then destroy all copies of the transmission.

**Comment by krychla1** [comment]
There is a SocketIOClientDotNet library, which is the closest to your bot, however I am not able to get any response from ED.

Every other single websocket library refuses HTTPS protocol. Is there any particular reason, that you do not use the websocket protocol (WS or WSS) for your api?

**Comment by zackcoburn** [comment]
I'll have to look into this.

Zack Coburn

NOTICE: This e-mail transmission (and/or the attachments accompanying it)
may contain confidential information belonging to the sender which is
protected by law. The information is intended only for the use of the
intended recipient. If you are not the intended recipient, you are hereby
notified that any disclosure, copying, distribution or the taking of any
action in reliance on the contents of this information is strictly
prohibited. Any unauthorized interception of this transmission is illegal.
If you have received this transmission in error, please promptly notify the
sender by reply e-mail, and then destroy all copies of the transmission.

**Comment by bennyprofane** [comment]
Just wanted to chime in that I'm having the same issue with python. `ValueError: scheme https is invalid` is the error that is returned when I try to connect to the current https socket. If you are able to switch to WS protocol the python libraries I'm using should work without issue. Thanks.

**Comment by lampshade9909** [comment]
Having the same issue.  I can only get this working in javascript, not any other language due to the https going against the ws or wss standard.

**Comment by nowkin** [comment]
also have problem get it working in c#
always get a 520 origin error when i am trying to connect
WebException: Der Remoteserver hat einen Fehler zurückgegeben: (520) Origin Error.

**Comment by nowkin** [comment]
with the SocketIOClientDotNet  lib in c# i am also running in the following error while trying to connect

Quobject.EngineIoClientDotNet.Client.EngineIOException: xhr poll error ---> System.AggregateException: Mindestens ein Fehler ist aufgetreten. ---> System.Net.WebException: Der Remoteserver hat einen Fehler zurückgegeben: (400) Ungültige Anforderung.
   bei System.Net.HttpWebRequest.GetResponse()
   bei Quobject.EngineIoClientDotNet.Client.Transports.PollingXHR.XHRRequest.<Create>b__7_0()
   bei System.Threading.Tasks.Task.InnerInvoke()
   bei System.Threading.Tasks.Task.Execute()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   bei System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   bei Quobject.EngineIoClientDotNet.Client.Transports.PollingXHR.XHRRequest.Create()

**Comment by wesolykapselek** [comment]
First of all try to connect to URL: wss://socket.etherdelta.com/socket.io/?transport=websocket

But I'm still not able to call certain endpoint like getMarket

**Comment by nowkin** [comment]
i managed now to get response from the api

`JObject out= JObject.FromObject(new { token = "", user = "" });
socket.Emit("getMarket", out);

socket.On("market",
                (market) =>
                {
                    dynamic json= JObject.Parse(market.ToString());
                });
`
now the only problem is to get only one token with the token adress. but this is also not working for me with socket.io in php

**Comment by wesolykapselek** [comment]
Thanks for the response.

But in java method sending json has only one argument which - probably - is the json as string.

I'm curious how "getMarket" part is mapped. It's attached at the end of the URL or something? Are you able to check this?

**Comment by nowkin** [comment]
no the getMarket will be emitted when you are connected with the socket
in my example socket.Emit

**Comment by wesolykapselek** [comment]
Yeah I know but what is getMarket? Method? URL part?

**Comment by nowkin** [comment]
a methodcall on the socket server

**Comment by hcancelik** [comment]
@wesolykapselek were you able to figure out? I'm in the same boat as you're. 

Thanks for the URL btw.

**Comment by wesolykapselek** [comment]
@hcancelik Nope. Still dead end. Stopped developing few days after above post. Tried few libraries and also socket.io port to Java. This socket.io had the worst result.

**Comment by hcancelik** [comment]
All I get is a response similar to the one below, using the URL you provided but couldn't move forward with getMarket method yet :(

`"0{"sid":"4JgLiJlrQ02g5lasdAvR","upgrades":[],"pingInterval":25000,"pingTimeout":60000}"`

**Comment by wesolykapselek** [comment]
@hcancelik  Yep. Ended with the same result.

**Comment by RomanBrodetski** [comment]
I face the same issue. I've tried several scala websocket libraries, and they all complain on the protocol not being `wss`.

**Comment by JonathonDunford** [comment]
So, I sent the following and it pulled the market for all tokens. Don't ask me why, but it worked.

42["getMarket",{"token":""}]

**Comment by zartdinov** [comment]
Looks like only `socket.io` library works fine for Node.js.

Unable to connect with `ws` or `websocket`.
